src/utils/goma.js

```diff
@@ -1,16 +1,92 @@
 const childProcess = require('child_process');
+const crypto = require('crypto');
 const fs = require('fs');
 const path = require('path');
+const rimraf = require('rimraf');
+const { unzipSync } = require('cross-zip');
 const { color } = require('./logging');
 
-const getExternalBinaries = root => path.resolve(root, 'src', 'electron', 'external_binaries');
-const gomaDirExists = root => fs.existsSync(path.resolve(getExternalBinaries(root), 'goma'));
+const gomaDir = path.resolve(__dirname, '..', '..', 'third_party', 'goma');
+const gomaGnFile = path.resolve(__dirname, '..', '..', 'third_party', 'goma.gn');
+const gomaShaFile = path.resolve(__dirname, '..', '..', 'third_party', 'goma', '.sha');
+const gomaBaseURL = 'https://electron-build-tools.s3-us-west-2.amazonaws.com/build-dependencies';
 
-function gomaIsAuthenticated(root) {
-  const gomaDir = path.resolve(getExternalBinaries(root), 'goma');
+const GOMA_PLATFORM_SHAS = {
+  darwin: '129235876fed7ad09973b3aab0c5b62c215cff227f56a776bbd838cbf2d9d549',
+  linux: '0d210c75cb39a0db88dea2c9ecca6d9607c0d8d7256be22eb39303130116ef0c',
+  win32: 'dbed01176cfc45adfe6c41045307cba1b0b4851ce6d9ad1f96beed180592924d',
+};
+
+const isSupportedPlatform = !!GOMA_PLATFORM_SHAS[process.platform];
+
+function downloadAndPrepareGoma() {
+  if (!isSupportedPlatform) return;
+
+  const gomaGnContents = `goma_dir = "${gomaDir}"\nuse_goma = true`;
+  if (!fs.existsSync(gomaGnFile) || fs.readFileSync(gomaGnFile, 'utf8') !== gomaGnContents) {
+    console.log(`Writing new goma.gn file ${color.path(gomaGnFile)}`);
+    fs.writeFileSync(gomaGnFile, gomaGnContents);
+  }
+  const sha = GOMA_PLATFORM_SHAS[process.platform];
+  if (
+    fs.existsSync(gomaShaFile) &&
+    fs.readFileSync(gomaShaFile, 'utf8') === sha &&
+    !process.env.ELECTRON_FORGE_GOMA_REDOWNLOAD
+  )
+    return;
+
+  const filename = {
+    darwin: 'goma-mac.tgz',
+    linux: 'goma-linux.tgz',
+    win32: 'goma-win.zip',
+  }[process.platform];
+
+  const tmpDownload = path.resolve(gomaDir, '..', filename);
+  // Clean Up
+  rimraf.sync(gomaDir);
+  rimraf.sync(tmpDownload);
+
+  const downloadURL = `${gomaBaseURL}/${sha}/${filename}`;
+  console.log(`Downloading ${color.cmd(downloadURL)} into ${color.path(tmpDownload)}`);
+  childProcess.spawnSync(
+    process.execPath,
+    [path.resolve(__dirname, '..', 'download.js'), downloadURL, tmpDownload],
+    {
+      stdio: 'inherit',
+    },
+  );
+  const hash = crypto
+    .createHash('SHA256')
+    .update(fs.readFileSync(tmpDownload))
+    .digest('hex');
+  if (hash !== sha) {
+    console.error(
+      `${color.err} Got hash for downloaded file ${color.cmd(hash)} which did not match ${color.cmd(
+        sha,
+      )}. Halting now`,
+    );
+    rimraf.sync(tmpDownload);
+    process.exit(1);
+  }
+
+  if (filename.endsWith('.tgz')) {
+    const result = childProcess.spawnSync('tar', ['zxvf', filename], {
+      cwd: path.resolve(tmpDownload, '..'),
```

Comment:
_nit_

we use `path.resolve(tmpDownload, '..')` in both the `if` and `else` clauses... maybe move it to the line before `if`

```js
const extractPath = path.dirname(tmpDownload);
if (...
```